core/src/test/java/io/github/tcdl/ResponderServerTest.java

```diff
@@ -1,23 +1,50 @@
 package io.github.tcdl;
 
-import io.github.tcdl.adapters.MockAdapter;
-import io.github.tcdl.config.MsbMessageOptions;
-import io.github.tcdl.messages.Message;
+import io.github.tcdl.events.Event;
+import io.github.tcdl.middleware.Middleware;
+import io.github.tcdl.middleware.MiddlewareChain;
 import io.github.tcdl.support.TestUtils;
-import io.github.tcdl.support.Utils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
 
-import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
 
 /**
  * Created by rdro on 4/30/2015.
  */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ ResponderServer.class, Responder.class } )
 public class ResponderServerTest {
 
-    // TODO write tests
     @Test
-    public void testResponderServer() throws Exception {
+    public void testResponderServerProcessSuccess() throws Exception {
+
+        Middleware middleware = (request, responder) -> {};
+
+        ResponderServer
+                .create(TestUtils.createSimpleConfig())
+                .use(middleware)
+                .listen();
+
+        mockStatic(CompletableFuture.class);
+        ArgumentCaptor<Supplier> middlewareCaptor = ArgumentCaptor.forClass(Supplier.class);
+
+        ChannelManager.getInstance().emit(Event.MESSAGE_EVENT, TestUtils.createMsbRequestMessageWithPayload());
```

Comment:
Why do we need to invoke another class to tests this one? Can't we invoke some public method on ResponderServer directly?